Editor/EditorCore/VertexManipulationTool.cs

```diff
@@ -267,6 +268,10 @@ protected void BeginEdit(string undoMessage)
 
             foreach (var mesh in selection)
             {
+                ShapeComponent shape;
+                if(mesh.gameObject.TryGetComponent(out shape))
+                    shape.edited = true;
```

Comment:
This isn't a reliable way to set the edited flag. In fact I don't know if we have a good way to do this at the moment that can accurately handle all cases. It may require us to add some sort of hash or ID to the `ProBuilderMesh` class that can be used to track changes.

Reply:
I Added this remark on the Shape tool's Jira